simulator/memory/elf/elf_loader.cpp

```diff
@@ -23,11 +24,11 @@ ElfLoader::ElfLoader( const std::string& filename, AddrDiff offset)
         throw InvalidElfFile( filename);
 }
 
-void ElfLoader::load_to( FuncMemory* memory) const
+void ElfLoader::load_to( std::shared_ptr<FuncMemory> memory) const
```

Comment:
My guideline would be to use a raw pointer here, as we are not going to own Memory inside ElfLoader.

Reply:
Alternatively, we may use `const std::shared_ptr<FuncMemory>&`. But reference to pointer is a awkward pattern...